File: ConduitLLM.Core/Exceptions/InvalidRequestException.cs
using System;

namespace ConduitLLM.Core.Exceptions
{
    /// <summary>
    /// Exception thrown when a request contains invalid parameters or is malformed.
    /// Maps to HTTP 400 Bad Request status code.
    /// </summary>
    public class InvalidRequestException : ConduitException
    {
        /// <summary>
        /// Gets the error code associated with this invalid request.
        /// </summary>
        public string? ErrorCode { get; }

        /// <summary>
        /// Gets the parameter that caused the validation error, if applicable.
        /// </summary>
        public string? Param { get; }

        /// <summary>
        /// Initializes a new instance of the <see cref="InvalidRequestException"/> class.
        /// </summary>
        /// <param name="message">The error message describing what is invalid.</param>
        public InvalidRequestException(string message) 
            : base(message)
        {
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="InvalidRequestException"/> class with an error code.
        /// </summary>
        /// <param name="message">The error message describing what is invalid.</param>
        /// <param name="errorCode">The specific error code for this type of invalid request.</param>
        public InvalidRequestException(string message, string errorCode) 
            : base(message)
        {
            ErrorCode = errorCode;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="InvalidRequestException"/> class with error code and parameter.
        /// </summary>
        /// <param name="message">The error message describing what is invalid.</param>
        /// <param name="errorCode">The specific error code for this type of invalid request.</param>
        /// <param name="param">The parameter that caused the validation error.</param>
        public InvalidRequestException(string message, string errorCode, string param) 
            : base(message)
        {
            ErrorCode = errorCode;
            Param = param;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="InvalidRequestException"/> class with an inner exception.
        /// </summary>
        /// <param name="message">The error message describing what is invalid.</param>
        /// <param name="innerException">The inner exception.</param>
        public InvalidRequestException(string message, Exception innerException) 
            : base(message, innerException)
        {
        }
    }
}

File: SDKs/Node/Core/src/errors/InvalidRequestException.ts
import { ConduitError } from '@knn_labs/conduit-common';

/**
 * Error thrown when a request contains invalid parameters or is malformed
 * Maps to HTTP 400 status code
 */
export class InvalidRequestException extends ConduitError {
  public readonly errorCode?: string;
  public readonly param?: string;

  constructor(message: string, errorCode?: string, param?: string) {
    super(
      message,
      400,
      errorCode ?? 'invalid_request',
      {
        type: 'invalid_request_error',
        errorCode,
        param
      }
    );
    this.errorCode = errorCode;
    this.param = param;
    this.name = 'InvalidRequestException';
  }
}

export function isInvalidRequestException(error: unknown): error is InvalidRequestException {
  return error instanceof InvalidRequestException ||
    (error instanceof ConduitError && error.statusCode === 400 && error.type === 'invalid_request_error');
}